scripts/validate_docstrings.py

```diff
@@ -60,6 +60,7 @@
 PRIVATE_CLASSES = ["NDFrame", "IndexOpsMixin"]
 DIRECTIVES = ["versionadded", "versionchanged", "deprecated"]
 DIRECTIVE_PATTERN = re.compile(rf"^\s*\.\. ({'|'.join(DIRECTIVES)})(?!::)", re.I | re.M)
+BULLET_LIST_PATTERN = re.compile(r"(?<=:)(\*|-|\+|•|‣|⁃)\s.*")
```

Comment:
I don't think we ever use anything but `-` and `*` for bullet points. Is this expression working?

I feel like it'd be simpler to simply check if the last line starts by `..`, `*`, `-` or is more indented than the rest of the docstring. If it does skip the check for ending with a period. I think this should capture most of the cases, if not all.

Reply:
Hi @datapythonista, thank your for all the suggestions.

I have included all those bullets points because it was suggested in a comment above.

Also I think I have implemented something similar to your suggestion in few commits before (everything before and including this commit
https://github.com/pandas-dev/pandas/pull/25786/commits/331e518387c161e73889164a70e348130fb4cfe1
)

Should I revert to this implementation?

Reply:
@WillAyd I guess you suggested the regex approach. Not sure if I'm missing something, but I feel it's not feasible to capture whether the end of the parameter description is a list or a directive, in a simple way. Is it numpydoc preprocessing the value and making it easy, or do you think we can detect line breaks since the bullet point...?

Reply:
I'm not tied to it if there's a better way. I just think the original incarnation here had too many loops, so fine with whatever strikes a readable balance you may know of

Reply:
Thanks @WillAyd, I agree that this shouldn't need a complex implementation with loops.

@EdAbati do you mind pasting here what is the value we get for the parameter description of the docstring in the `parameters_with_bullet_points1` test?

My guess is that we can just validate that the last line doesn't start with:
- `*` or `-`
- `..`
- 4 spaces

If it doesn't, we require the final period, if it does, we don't require it.

Does it make sense?

Reply:
Hi both,

@datapythonista the `parameter_desc` method applied to `parameters_with_bullet_points1` returns:
```
OrderedDict([('method',
              ("{'min', 'max'}, default 'min'",
               "Parameter introduction:* 'min': the description for min* 'max': the description for max that can also be very long and  requires an extra row"))])
```
I thought to use the `.doc["Parameters"] ` method of the `NumpyDocString` class just so we could access the parameter description line by line (as a list) and assess the last line.

Reply:
I guess we join all the lines ourselves, I guess we can return the list, or however numpydoc returns this, and just join where we need it (I guess we don't use this description in many places).

Btw, we're moving this script to numpydoc, we'll have to make this changes there, once https://github.com/numpy/numpydoc/pull/238 is merged (quite soon).

Reply:
Ok, no problem I will close this for now. Thank you for the useful feedback